label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleValidatorImpl.java

```diff
@@ -127,21 +129,43 @@ private void validateUnaryCondition(UnaryCondition unaryCondition) {
   }
 
   private void validateAction(Action action) {
-    validateNonDefaultPresenceOrThrow(action, action.ENTITY_TYPE_FIELD_NUMBER);
+    validateNonDefaultPresenceOrThrow(action, action.ENTITY_TYPES_FIELD_NUMBER);
+    validateEntityTypes(action.getEntityTypesList());
     validateNonDefaultPresenceOrThrow(action, action.OPERATION_FIELD_NUMBER);
     switch (action.getValueCase()) {
-      case STATIC_LABEL_ID:
-        validateNonDefaultPresenceOrThrow(action, action.STATIC_LABEL_ID_FIELD_NUMBER);
+      case STATIC_LABELS:
+        validateStaticLabels(action.getStaticLabels());
         break;
       case DYNAMIC_LABEL:
         validateDynamicLabel(action.getDynamicLabel());
         break;
+      case DYNAMIC_LABEL_KEY:
+        validateNonDefaultPresenceOrThrow(action, action.DYNAMIC_LABEL_KEY_FIELD_NUMBER);
+        break;
       default:
         throwInvalidArgumentException(
             String.format("Unexpected Case in %s:%n %s", getName(action), printMessage(action)));
     }
   }
 
+  void validateEntityTypes(List<String> entityTypes) {
+    Set<String> entityTypesSet = new HashSet<>(entityTypes);
+    if (entityTypesSet.size() < entityTypes.size()) {
```

Comment:
minor nit: One less variable name and one less potential bug (that is, it's easy to flip the >, while != is symmetric and gives us the same info)
```suggestion
    if (Set.copyOf(entityTypes).size() != entityTypes.size()) {
```

Reply:
done.